src/OpenTelemetry.Sampler.AWS/RulesCache.cs

```diff
@@ -109,6 +109,66 @@ public SamplingResult ShouldSample(in SamplingParameters samplingParameters)
         return this.FallbackSampler.ShouldSample(in samplingParameters);
     }
 
+    public List<SamplingStatisticsDocument> Snapshot(DateTime now)
+    {
+        List<SamplingStatisticsDocument> snapshots = new List<SamplingStatisticsDocument>();
+        foreach (var ruleApplier in this.RuleAppliers)
```

Comment:
I am not as familiar with c#.
Would there be an issue with iterating over this list and editing it at the same time? In Java this could throw a concurrent modification exception.

Reply:
I se, thanks for the explanation!